### PR TITLE
Idex 3790 Improve the menu structure

### PR DESCRIPTION
**Pull request by vitaliy-guliy**

### CODE REVIEW

File: ide/che-core-ide-app/src/main/java/org/eclipse/che/ide/menu/MenuResources.java [hunk 1]
```diff
@@ -17,7 +17,7 @@
  * @author Evgen Vidolob
  */
 public interface MenuResources extends ClientBundle {
-    public interface Css extends CssResource {
+    interface Css extends CssResource {
```
Comment (sleshchenko) [inline ide/che-core-ide-app/src/main/java/org/eclipse/che/ide/menu/MenuResources.java]: Can you explain why do you remove public modifier?

File: ide/che-core-ide-app/src/main/java/org/eclipse/che/ide/actions/CreateProjectAction.java [hunk 2]
```diff
@@ -30,15 +30,15 @@
...
 
     private final ProjectWizardPresenter wizard;
     private final AnalyticsEventLogger   eventLogger;
     private final AppContext             appContext;
 
     @Inject
-    public NewProjectAction(Resources resources, ProjectWizardPresenter wizard, AnalyticsEventLogger eventLogger, AppContext appContext) {
-        super(Arrays.asList(PROJECT_PERSPECTIVE_ID), "Project...", "Create new project", null, resources.newProject());
+    public CreateProjectAction(Resources resources, ProjectWizardPresenter wizard, AnalyticsEventLogger eventLogger, AppContext appContext) {
+        super(Arrays.asList(PROJECT_PERSPECTIVE_ID), "Create Project...", "Create new project", null, resources.newProject());
```
Comment (azatsarynnyy) [inline ide/che-core-ide-app/src/main/java/org/eclipse/che/ide/actions/CreateProjectAction.java]: I propose to move action's title and description to CoreLocalizationConstants

Comment (vparfonov) [inline ide/che-core-ide-app/src/main/java/org/eclipse/che/ide/actions/CreateProjectAction.java]: +1

File: ide/che-core-ide-app/src/main/java/org/eclipse/che/ide/core/StandardComponentInitializer.java [hunk 3]
```diff
@@ -359,42 +355,227 @@ public void initialize() {
...
-        importProjectGroup.addAction(importLocalProjectAction);
+        // Workspace (New Menu)
+        DefaultActionGroup workspaceGroup = (DefaultActionGroup)actionManager.getAction(IdeActions.GROUP_WORKSPACE);
 
-        // Compose New group
+        //  -- Import Project (Note: launches the Import From Location popup, no more submenu)
+        actionManager.registerAction("importProject", importProjectAction);
+        workspaceGroup.add(importProjectAction);
+
+        //  -- Create Project (Previously: File -> New -> Project)
```
Comment (azatsarynnyy) [inline ide/che-core-ide-app/src/main/java/org/eclipse/che/ide/core/StandardComponentInitializer.java]: I think such comments are useless

File: ide/che-core-ide-app/src/main/java/org/eclipse/che/ide/core/StandardComponentInitializer.java [hunk 4]
```diff
@@ -359,42 +355,227 @@ public void initialize() {
...
+                event.getPresentation().setVisible(true);
+                event.getPresentation().setEnabled(false);
+            }
+
+            @Override
+            public void actionPerformed(ActionEvent e) {
+            }
+        });
+
+        //  -- Create Factory...  (This is for Codenvy-only, not in Che)
```
Comment (vparfonov) [inline ide/che-core-ide-app/src/main/java/org/eclipse/che/ide/core/StandardComponentInitializer.java]: Can we move it to factory plugin?

File: ide/che-core-ide-app/src/main/java/org/eclipse/che/ide/core/StandardComponentInitializer.java [hunk 5]
```diff
@@ -359,42 +355,227 @@ public void initialize() {
...
+                event.getPresentation().setVisible(true);
+                event.getPresentation().setEnabled(false);
+            }
+
+            @Override
+            public void actionPerformed(ActionEvent e) {
+            }
+        });
+
+        //  -- Clone...           (This is for Codenvy-only, not in Che)
```
Comment (vparfonov) [inline ide/che-core-ide-app/src/main/java/org/eclipse/che/ide/core/StandardComponentInitializer.java]: Can we move it to factory plugin?

File: ide/che-core-ide-api/src/main/java/org/eclipse/che/ide/api/action/AbstractPerspectiveAction.java [hunk 6]
```diff
@@ -27,15 +27,19 @@
  */
 public abstract class AbstractPerspectiveAction extends Action {
 
-    private final List<String> activePerspectives;
+    /**
+     * A list of perspectives in which the action is enabled.
+     * Null or empty list means the action is enabled everywhere.
```
Comment (vzhukovskii) [inline ide/che-core-ide-api/src/main/java/org/eclipse/che/ide/api/action/AbstractPerspectiveAction.java]: Disagree with that. AbstractPerspectiveAction was created only to show action in specific perspectives. If you want to create action that will be shown everywhere you can create action that simply extends Action.

File: ide/che-core-ide-app/src/main/java/org/eclipse/che/ide/menu/MenuResources.java (hunk 1 above)
Comment (azatsarynnyy) [inline ide/che-core-ide-app/src/main/java/org/eclipse/che/ide/menu/MenuResources.java]: Oh, common ) It's public by default.

File: ide/che-core-ide-app/src/main/java/org/eclipse/che/ide/core/StandardComponentInitializer.java [hunk 7]
```diff
@@ -359,42 +355,227 @@ public void initialize() {
...
+                event.getPresentation().setEnabled(false);
+            }
+
+            @Override
+            public void actionPerformed(ActionEvent e) {
+            }
+        });
+
+        //  -- Stop... (Previously: Machine -> Stop Workspace)
+        workspaceGroup.add(new AbstractPerspectiveAction(null, "Stop...", null, null, null) {
```
Comment (vzhukovskii) [inline ide/che-core-ide-app/src/main/java/org/eclipse/che/ide/core/StandardComponentInitializer.java]: What the profit of dummy action?

File: ide/che-core-ide-app/src/main/java/org/eclipse/che/ide/core/StandardComponentInitializer.java [hunk 8]
```diff
@@ -405,61 +586,15 @@ public void initialize() {
...
-        // Compose Help menu
-        DefaultActionGroup helpGroup = (DefaultActionGroup)actionManager.getAction(IdeActions.GROUP_HELP);
-        actionManager.registerAction("findActionAction", findActionAction);
-        actionManager.registerAction("hotKeysList", hotKeysListAction);
+        helpGroup.add(showPreferencesAction);
 
-        helpGroup.add(findActionAction);
-        helpGroup.add(hotKeysListAction);
-        helpGroup.addSeparator();
+        actionManager.registerAction("setupProjectAction", settingsAction);
```
Comment (vzhukovskii) [inline ide/che-core-ide-app/src/main/java/org/eclipse/che/ide/core/StandardComponentInitializer.java]: Why "setupProjectAction"?

File: ide/che-core-ide-ui/src/main/java/org/eclipse/che/ide/ui/toolbar/MenuLockLayer.java [hunk 9]
```diff
@@ -67,21 +67,20 @@ public MenuLockLayer(CloseMenuHandler closeMenuCallback, int topOffset) {
...
-        setWidth("" + width + "px");
-        setHeight("" + height + "px");
-        DOM.setElementAttribute(getElement(), "id", "menu-lock-layer-id");
-        DOM.setStyleAttribute(getElement(), "zIndex", "" + (Integer.MAX_VALUE - 5));
+        getElement().getStyle().setProperty("right", "0px");
+        getElement().getStyle().setProperty("bottom", "0px");
+        getElement().getStyle().setProperty("zIndex", (Integer.MAX_VALUE - 5) + "");
 
         AbsolutePanel blockMouseEventsPanel = new LockLayer();
         blockMouseEventsPanel.setStyleName("exo-lockLayer");
```
Comment (vzhukovskii) [inline ide/che-core-ide-ui/src/main/java/org/eclipse/che/ide/ui/toolbar/MenuLockLayer.java]: we need to rename this

Comment (vitaliy-guliy) [inline ide/che-core-ide-ui/src/main/java/org/eclipse/che/ide/ui/toolbar/MenuLockLayer.java]: rename in what?

File: ide/che-core-ide-app/src/main/java/org/eclipse/che/ide/core/StandardComponentInitializer.java (hunk 8 above)
Comment (vitaliy-guliy) [inline ide/che-core-ide-app/src/main/java/org/eclipse/che/ide/core/StandardComponentInitializer.java]: I don't know. You can ask an author of SettingsAction about that.

File: ide/che-core-ide-api/src/main/java/org/eclipse/che/ide/api/action/AbstractPerspectiveAction.java (hunk 6 above)
Comment (vitaliy-guliy) [inline ide/che-core-ide-api/src/main/java/org/eclipse/che/ide/api/action/AbstractPerspectiveAction.java]: Ok

File: ide/che-core-ide-ui/src/main/java/org/eclipse/che/ide/ui/toolbar/MenuLockLayer.java (hunk 9 above)
Comment (vzhukovskii) [inline ide/che-core-ide-ui/src/main/java/org/eclipse/che/ide/ui/toolbar/MenuLockLayer.java]: `exo-lockLayer` -> `che-lockLayer` or smth else

File: ide/che-core-ide-app/src/main/java/org/eclipse/che/ide/actions/CreateProjectAction.java (hunk 2 above)
Comment (vitaliy-guliy) [inline ide/che-core-ide-app/src/main/java/org/eclipse/che/ide/actions/CreateProjectAction.java]: Sorry guys, I have no more time for refactoring commands. I hope each coder will write perfect and well commented code.

File: ide/che-core-ide-app/src/main/java/org/eclipse/che/ide/core/StandardComponentInitializer.java (hunk 7 above)
Comment (vitaliy-guliy) [inline ide/che-core-ide-app/src/main/java/org/eclipse/che/ide/core/StandardComponentInitializer.java]: We have to show new disabled item in menu

Comment (vzhukovskii) [inline ide/che-core-ide-app/src/main/java/org/eclipse/che/ide/core/StandardComponentInitializer.java]: This action do nothing

**Pull request merged**